### ISSUE TITLE
Allow exceptions for style-prop-object

### ISSUE THREAD

**Issue by remcohaszing** [yannickcr/eslint-plugin-react#1813]
I am using the airbnb style guide, which enables the `react/style-prop-object` rule. For most use cases this is fine. However, I just started implementing [`react-intl`](https://github.com/yahoo/react-intl). This exposes a [`<FormattedNumber>`](https://github.com/yahoo/react-intl/wiki/Components#formattednumber) component which accepts a `style` prop. This triggers the `react/style-prop-object` rule.

I think it’s generally a bad practice to create props using such standarized names, but third party libraries might. I suggest adding an option to this rule, which defines an array of component names for which different style prop types are allowed.

For example:

```json
{
  "rules": {
    "react/style-prop-object": [
      "error",
      {"allow": ["FormattedNumber"]}
    ]
  }
}
```

**Comment by jseminck** [comment]
I'll take a stab at this issue.
